executor/load_data.go

```diff
@@ -330,13 +330,11 @@ func (e *LoadDataInfo) initLoadColumns(columnNames []string) error {
 		for _, v := range e.ColumnAssignments {
 			columnNames = append(columnNames, v.Column.Name.O)
 		}
+	}
 
-		cols, missingColName = table.FindCols(tableCols, columnNames, e.Table.Meta().PKIsHandle)
-		if missingColName != "" {
-			return errors.Errorf("LOAD DATA INTO %s: unknown column %s", e.Table.Meta().Name.O, missingColName)
```

Comment:
seems this function `initLoadColumns` handles the LOAD DATA ... SET col=xxx part, and `initFieldMappings` handles LOAD DATA ... (col1, col2) ... part. Maybe we can move the check to `initFieldMappings`.

Reply:
What benefit would that give us?

Reply:
not important. I tested the behaviour of different places the unknown column appears in MySQL:

```
CREATE TABLE `t` (
  `c` int(11) DEFAULT NULL,
  `c2` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1
```

```
MySQL [test]> load data local infile '/tmp/t.csv' into table t fields terminated by ',' (@v1, xxx1) set c=xxx2, c2=@v1;
ERROR 1054 (42S22): Unknown column 'xxx1' in 'field list'
```

although `c` is the first column in table and it's assigned from `xxx2`, MySQL thinks the first unknown column is `xxx1`. I guess we should process `... (col1, col2, ...) ...` part before `... SET col=xxx` and separately. Not sure that's over-imitation.


And the error message should also be changed.

Reply:
I fixed the message, it now looks like this:

Reply:
I've created #41468 for another case with a different error message and error number.